modules/log-analytics-workspace/vars.tf

```diff
@@ -28,5 +28,5 @@ variable "retention_in_days" {
 variable "tags" {
   description = "A mapping of tags to assign to the resource."
   type        = map(string)
-  default     = {}
+  default     = null
```

Comment:
Does this not cause a persistent diff like the other azure resources that accept tags? Most of their resources add an empty object after the apply has completed so we always see changes on every subsequent apply.

Reply:
Yea, `null` falls back to the provider defaults. We can use it to assert optional values in modules.
I haven't noticed persistent diff with tags in `azurerm`, i've seen it in `aws` though.

Reply:
We have many azurerm resources that are not yet passing tags, while they are supported. In each of those there is a persistent diff after a bit of time (I think that azure might have a process that comes behind and changes the resource attributes). If I recall correctly we are passing a default tag for this module from infrastructure-modules, but it will more than likely be removed in order to remove our branding and rely on infrastructure-live to pass the tags, for which we will not always be in control of. If there are not any tags then I'm pretty sure that there will be a persistent diff unless we default to an empty object, but I have not tested with this specific resource.

Reply:
Will there be a diff, even if we default to an empty object in [infrastructure-modules](https://github.com/quantum-sec/infrastructure-modules/pull/201/files#diff-a77291334582aeb6ef138a6db49a560dd5d02fe8261d21e826163793fc966ea8R55). Or should we be doing it, there as well as in the underlying module?

Reply:
We should not have the diff if we default there, but other users that may utilize this module outside of our implementation may. We should just ensure that we understand what will happen and either document the side-effect or default to not have a side-effect.

Reply:
This conversation, while focused on tags, really covers all defaults with `null`. We just need to understand what will happen when terraform passes `null` and gets back a value that is different - and when terraform queries for the next plan and gets back a value that is different from what is stored in remote state.

Reply:
Sure, will track the changes for this as a bug. Let's also get on a call tmrw to test out the same.